irc_send.c

```diff
@@ -293,15 +287,26 @@ void irc_send_who(irc_t *irc, GSList *l, const char *channel)
 	gboolean is_channel = strchr(CTYPES, channel[0]) != NULL;
 
 	while (l) {
-		irc_user_t *iu = l->data;
+		irc_user_t *iu;
+
+		/* Null terminated string with three chars, respectively:
+		 * { <H|G>, <@|%|+|\0>, \0 } */
+		char status_prefix[3] = {0};
```

Comment:
Yeah, well, quoting commit log:

>We can't actually have multiple prefixes internally

Modes/prefixes in bitlbee are actually just a 1:1 mapping from user status (online, away, special, offline) to a single character, determined by the `show_users` channel setting (which is a string like `online@,special%,away+,offline`, defaults to `online+,away`). That's in the case of control channels.

Chat channels don't actually support modes other than showing yourself and the root user as @. Even if they supported modes (to show who's a moderator in a groupchat, and i really want to implement this soonish), I don't think any IM protocols other than IRC support multiple statuses.

An earlier version of this patch had actual support for multiple prefixes:

http://dump.dequis.org/XENN4.txt (oh look, it almost has your name as the hash, neat)

...but i threw it away once i got to test it, because I couldn't find any way to set both flags at the same time. 

Either way, thanks for looking!